Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,370 @@
+# Bazel for iOS Developers
+
+This document is an introduction to Bazel for iOS developers. While there's
+been many docs written about Bazel, the existing material can be overwhelming
+for a line engineer looking to get started. This document is geared towards an
+iOS developer coming from Xcode and intends to be a lightweight introduction to
+Bazel.
+
+The document supplements canonical Bazel resources:
+- [Bazel overview](https://docs.bazel.build/versions/master/bazel-overview.html)
+- [A tutorial for iOS apps](https://docs.bazel.build/versions/master/tutorial/ios-app.html) 
+- [Bazel Concepts and terminology](https://docs.bazel.build/versions/master/build-ref.html)
+- [Getting started with Bazel](https://docs.bazel.build/versions/master/getting-started.html) 
+
+## Build systems for iOS developers
+
+First, let's address the definiton of a `Build system`. According to [Stack
+overflow](https://stackoverflow.com/questions/7249871/what-is-a-build-tool_),
+_Build tools are programs that automate the creation of executable applications
+from source code(eg. .apk for android app). Building incorporates
+compiling,linking and packaging the code into a usable or executable form._
+
+In iOS development, Apple encapsulates build systems inside of Xcode. Xcode is
+both the IDE _and build system_. For many developers, Xcode just works. To have
+a good experience, most developers don't need to worry about implementing build
+systems, IDEs, or compilers. When the project scales, developer experience
+drops off. Tasks like building, indexing, or merging a change that adds a new
+file can become painful. 
+
+Bazel makes it easy to manage build configuration in a way that results are
+functional and reproducible. In addition to producing an iOS application, Bazel
+makes it easy to automate many other kinds tasks for example, code generating a
+thrift schema, generating an Xcode project, and pushing a docker container.
+Having the ability to optimize the build system can significantly improve the
+developer experience.
+
+Compared to other mainstream build systems, Bazel is strict about inputs and
+outputs. This characteristic makes it reproducible and well suited for
+distributed cloud builds and caching. _To learn about how Bazel compares to
```

Comment:
Its probably helpful to add how strictness or cloud builds improve the developer experience.

Reply:
Good point, here's a few I can think of:
- performance ( this one is a pretty big one ).
- not having to constantly remove derived data for local builds
- makes it easier to manage CI with caching ( e.g. we haven't cleaned it since adopting Bazel )

Overall, it'd probably be good to have a concise segment on the benefits